src/Pages/OfficerDB/Edits/Editor.js

```diff
@@ -0,0 +1,160 @@
+import React, { useState } from 'react'
+import {
+  Modal,
+  ModalHeader,
+  ModalBody,
+  Dropdown,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem,
+  Input
+} from 'reactstrap'
+import ConfirmModule from './ConfirmationModule'
+import Uploader, { upload, getFileName } from './Uploader'
+import {
+  deletePicture,
+  editOfficer,
+  editAccessLevel
+} from '../../../APIFunctions/OfficerDB'
+
+const enums = require('../../../Enums')
+
+export default function Editor (props) {
+  const [roleDropdown, setRoleDropdown] = useState(false)
+  const [role, setRole] = useState(props.officer.role)
+  const [accessLevel, setAccessLevel] = useState(props.officer.accessLevel)
+  const [linkedIn, setLinkedIn] = useState('')
+  const [github, setGithub] = useState('')
+  const [quote, setQuote] = useState('')
+  const [picture] = useState(props.officer.pictureName)
+  const optionalInputs = [
+    {
+      placeholder: 'New linkedIn: ' + props.officer.linkedIn,
+      onChange: event => setLinkedIn(event.target.value)
+    },
+    {
+      placeholder: 'New github: ' + props.officer.github,
+      onChange: event => setGithub(event.target.value)
+    },
+    {
+      placeholder: 'New quote: ' + props.officer.quote,
+      onChange: event => setQuote(event.target.value)
+    }
+  ]
+
+  async function handleSubmitOfficer () {
+    let newPic = ''
+    // update image if change
+    if (getFileName()) {
+      deletePicture(picture)
+      newPic = getFileName()
+      // upload picture
+      upload()
+    }
+
+    const officer = props.officer && {
+      pictureName: newPic || picture,
+      name: props.officer.name,
+      email: props.officer.email,
+      accessLevel: accessLevel,
+      linkedIn: linkedIn,
+      github: github,
+      quote: quote,
+      role: role
+    }
+    // Delete empty fields
```

Comment:
why are you deleting it? you should just handle empty fields in your API [like this](https://github.com/SCE-Development/Core-v4/blob/89fbac4635e7ae72a54cc8d2a2cced59f071e708/api/routes/Event.js#L69)